### PR TITLE
Implement Post admin panel

### PR DESCRIPTION
**Pull request by teguhbayu**

### CODE REVIEW

File: src/app/(admin)/admin/posts/_components/ImageModal.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { H3 } from "@/app/_components/global/Text";
+import { FaX } from "react-icons/fa6";
+import { TextField } from "@/app/_components/global/Input";
+import { LinkWithCountAndUser } from "@/types/entityRelations";
+import { Dispatch, SetStateAction } from "react";
+import Form from "./parts/ImageForm";
+// import { updateLink } from "../action";
+// import FormButton from "./part/SubmitButton";
```
Comment (justpiple) [inline src/app/(admin)/admin/posts/_components/ImageModal.tsx]: unused code?

File: src/app/(admin)/admin/posts/_components/Table.tsx [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+import { PostWithTagsAndUser } from "@/types/entityRelations";
+import DataTable, { TableColumn } from "react-data-table-component";
+import { stringifyCompleteDate } from "@/utils/atomics";
+import { useEffect, useState } from "react";
+import { useRouter } from "next-nprogress-bar";
+import { toast } from "sonner";
+import { FaRegCopy, FaRegTrashAlt } from "react-icons/fa";
+import { MdPublish, MdUnpublished } from "react-icons/md";
+import { postDelete, updatePostStatus } from "../action";
+// import { cloneForm, deleteForm } from "../actions";
```
Comment (justpiple) [inline src/app/(admin)/admin/posts/_components/Table.tsx]: unused code?

File: src/app/(admin)/admin/posts/_components/Table.tsx [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+  async function updateStatus(state: boolean, id: string) {
+    if (!confirm("Apakah anda ingin mengupdate post ini?")) return;
+    const toastId = toast.loading("Loading...");
+    const action = await updatePostStatus(state, id);
+    if (action.error) return toast.error(action.message, { id: toastId });
+    setTimeout(() => {
+      toast.success(action.message, { id: toastId });
+    }, 3000);
+    router.refresh();
+    // router.push(`/admin/form/${action.data?.id}`);
```
Comment (justpiple) [inline src/app/(admin)/admin/posts/_components/Table.tsx]: unused code?

**Review by justpiple — APPROVED** [review]
OK good

**Comment by teguhbayu** [comment]
makasih om @justpiple 😘 i lop yu ❣️

**Pull request merged**